### ISSUE TITLE
History path lost on Safari while using router.base on nuxt.config.js

### ISSUE THREAD

**Issue by Meeeeow** [nuxt/nuxt.js#2638]
Here are steps and requirements to reproduce:
- Any iOS device, tested on iOS11/10/9
- Nuxt 1.1.1

1. Install `nuxt-starter-kit` template
2. Add router configure in to nuxt.config.js ↓
```js
router: {
  base: '/abc'
}
```
3. Create `a.vue` in `pages` directory, add `<nuxt-link to="/b">To Page B</nuxt-link>` in template
4. Create `b.vue` in `pages` directory, add any text into template
5. Open "http://you-host:your-port/abc/a"
6. Press "To Page B" button, after page loaded, press the back button

Actual result: It return to "http://your-host:your-port/abc"
Excepted result: Should return to last page "http://your-host:your-port/abc/a"

**Comment by llpw4** [comment]
I met the same problem;  In addition, in some android webview(like huawei P10 own browser), If you input  "http://you-host:your-port/abc/a" in the address  bar

**Comment by pi0** [comment]
Router base is incorrect. It should ends with a slash:
base: '/abc/'

**Comment by Meeeeow** [comment]
@pi0 I have two nuxt.js project with different configure, another one using `base: '/'` with same problem, it works in old release

**Comment by llpw4** [comment]
node_modules\nuxt\lib\app\router.js, in this file, find "createRouter" function,  delete "scrollBehavior ," , and  restart,  and then the problem is resolved.

**Comment by zhouyu1993** [comment]
@pi0 Yes, this problem is caused by `base: '/abc/'`. It will write `<base href="/abc/">` in `<head>`.
So, when `history.back()` it will go to `/abc/`.

**Comment by LDQ-first** [comment]
@llpw4 Thanks. It works for me. I want to know the reason.

**Comment by zhouyu1993** [comment]
@ZooTopiaGG Just try to delete `<base href="xxxx">` in `index.html` after `npm run build`.

**Comment by htfire** [comment]
I met the same problem, wish for the solution urgently;

**Comment by y-tsuno** [comment]
i have same problem.
Is it difficult to behavior in a subdirectory? (or  I shuold be give up "Browser Back" on Safari)

**Comment by luanwulin** [comment]
I come out a workaround
manually add a replaceState in `mounted` lifecycle
```javascript
       mounted(){
            // 解决ios丢失上一级url问题
            window.history.replaceState(null, null, this.$router.fullpath);
        }
```
this works fo me

**Comment by hqqxxf** [comment]
@llpw4 me too

**Comment by hqqxxf** [comment]
@luanwulin  it must added in every page, so terrible...

**Comment by shiyuegongsui** [comment]
me too~~

**Comment by luanwulin** [comment]
@hqqxxf you may add this in default.vue

**Comment by hqqxxf** [comment]
@luanwulin can you tell me why it works?

**Comment by luanwulin** [comment]
@hqqxxf sorry, i dont know either
